src/src/Controller/AbandonedCartController.php

```diff
@@ -0,0 +1,248 @@
+use PrestaShop\Module\SmailyForPrestaShop\Lib\Logger;
+use PrestaShop\Module\SmailyForPrestaShop\Model\AbandonedCart;
+use PrestaShop\Module\SmailyForPrestaShop\Model\AbandonedCartCollection;
+use PrestaShop\Module\SmailyForPrestaShop\Model\SentCart;
+use PrestaShop\PrestaShop\Core\ConfigurationInterface;
+
+class AbandonedCartController
+{
+    /**
+     * @var ConfigurationInterface
+     */
+    private $configuration;
+
+    /**
+     * @var Api;
+     */
+    private $api;
+
+    public function __construct(ConfigurationInterface $configuration)
+    {
+        $this->configuration = $configuration;
+
+        $subdomain = $configuration->get('SMAILY_SUBDOMAIN');
+        $username = $configuration->get('SMAILY_USERNAME');
+        $password = $configuration->get('SMAILY_PASSWORD');
+
+        if (!empty($subdomain) && !empty($username) && !empty($password)) {
+            $this->api = new Api($subdomain, $username, $password);
+        }
+    }
+
+    public function sync(string $token): bool
+    {
+        if ($this->configuration->get('SMAILY_CART_CRON_TOKEN') !== $token) {
+            echo 'Access denied!';
+
+            return false;
+        }
+
+        if (!$this->configuration->getBoolean('SMAILY_ENABLE_ABANDONED_CART')) {
+            echo 'Abandoned cart disabled!';
+
+            return false;
+        }
+
+        if (empty($this->api)) {
+            echo 'Account setup not finished!';
+
+            return false;
+        }
+
+        $count = 0;
+        $failCount = 0;
+        $success = true;
+
+        $abandonedCartsCollection = new AbandonedCartCollection();
+        foreach ($abandonedCartsCollection->carts as $cart) {
+            // Notice!
+            // When running cron job in the same browser the session is shared
+            // and the abandoned cart `date_upd` time is renewed on each request.
+            // This doesn't allow the check to pass. I recommend running the cron job
+            // in a private window to avoid this.
+            if (!$this->isDelayTimePassed($cart)) {
+                break;
+            }
+
+            if (!$this->send($cart)) {
+                ++$failCount;
+                $success = false;
+            }
+
+            ++$count;
+        }
+
+        echo sprintf("%s abandoned cart email(s) sent!\r\n", $count);
+        if (!$success) {
+            echo sprintf("%s failed cart(s), check logs!\r\n", $failCount);
+        }
+
+        return $success;
+    }
+
+    private function isDelayTimePassed(AbandonedCart $cart): bool
+    {
+        $cart_updated_time = strtotime($cart->dateUpdated);
+        $syncInterval = $this->configuration->get('SMAILY_ABANDONED_CART_TIME');
+
+        $reminder_time = strtotime('+' . $syncInterval . ' minutes', $cart_updated_time);
+        $current_time = strtotime(date('Y-m-d H:i') . ':00');
+        // Don't continue if cart delay time has not passed.
+        if ($current_time < $reminder_time) {
```

Comment:
return $current_time >= $reminder_time;

Reply:
The idea is have as less nesting as possible.

Reply:
Fixed https://github.com/sendsmaily/smaily-prestashop-module/pull/57/commits/04a8e137ba83825cb9a949ba49adbc7e2974532e